sample_client/src/main/java/com/owncloud/android/lib/sampleclient/intropage.java

```diff
@@ -0,0 +1,247 @@
+                        @Override
+                        public void onClick(DialogInterface dialogInterface, int i) {
+                            dialogInterface.dismiss();
+                        }
+                    })
+                    .create()
+                    .show();
+            //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+            //                                          int[] grantResults)
+            // to handle the case where the user grants the permission. See the documentation
+            // for ActivityCompat#requestPermissions for more details.
+            //return;
+        }
+
+*/
+
+
+
+    public void addDotsIndicator(int position){
+        mDots=new TextView[sliderAdapterIntro.getCount()];
+        mDotsLayout.removeAllViews();
+        for (int i=0;i<mDots.length;i++){
+            mDots[i]=new TextView(this);
+            mDots[i].setText(Html.fromHtml("&#8226"));
+            mDots[i].setTextSize(25f);
+            mDots[i].setTextColor(getResources().getColor(R.color.colorAccent));
+            mDotsLayout.addView(mDots[i]);
+        }
+        if(mDots.length>0){
+            mDots[position].setTextColor(getResources().getColor(R.color.colorPrimaryDark));
+        }
+    }
+
+    ViewPager.OnPageChangeListener viewListner=new ViewPager.OnPageChangeListener() {
+        @Override
+        public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
+        }
+
+        @Override
+        public void onPageSelected(int i) {
+            addDotsIndicator(i);
+            currentPage=i;
+
+            if(i==0){
+                next_finish.setEnabled(true);
+                back.setEnabled(false);
+                back.setVisibility(View.INVISIBLE);
+
+                next_finish.setText("Next");
+                back.setText("");
+            }
+            else if(i==mDots.length-1){
+                next_finish.setEnabled(true);
+                back.setEnabled(true);
+                back.setVisibility(View.VISIBLE);
+
+                next_finish.setText("Finish");
+                back.setText("Back");
+            }
+            else {
+                next_finish.setEnabled(true);
+                back.setEnabled(true);
+                back.setVisibility(View.VISIBLE);
+
+                next_finish.setText("Next");
+                back.setText("Back");
+
+            }
+        }
+
+        @Override
+        public void onPageScrollStateChanged(int state) {
+
+        }
+    };
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+
+
+//        final FirebaseAuth mAuth=FirebaseAuth.getInstance();
+        String introShow=prefs.getString("isIntroShow","");
+
+        if(introShow.equals("")) {
+            //  Toast.makeText(this, "", Toast.LENGTH_SHORT).show();
```

Comment:
how are you going to handle it if introShow is empty?

Reply:
Done with it by providing values